common/storage/sql/sqlite3/storage.c

```diff
@@ -950,6 +950,139 @@ retcode_t iota_stor_transaction_approvers_count(connection_t const* const conn,
   return ret;
 }
 
+static void iota_stor_transaction_find_in_clause(char* const statement,
+                                                 size_t* const offset,
+                                                 size_t count) {
+  *offset += sprintf(statement + *offset, "IN(");
+  for (size_t i = 0; i < count; i++) {
+    *offset += sprintf(statement + *offset, "?,");
+  }
+  statement[*offset - 1] = ')';
+}
+
+retcode_t iota_stor_transaction_find(connection_t const* const conn,
+                                     hash243_queue_t const bundles,
+                                     hash243_queue_t const addresses,
+                                     hash81_queue_t const tags,
+                                     hash243_queue_t const approvees,
+                                     iota_stor_pack_t* const pack) {
+  retcode_t ret = RC_OK;
+  sqlite3_stmt* sqlite_statement = NULL;
+  size_t bundles_count = hash243_queue_count(bundles);
+  size_t addresses_count = hash243_queue_count(addresses);
+  size_t tags_count = hash81_queue_count(tags);
+  size_t approvees_count = hash243_queue_count(approvees);
+  hash243_queue_entry_t* iter243 = NULL;
+  hash81_queue_entry_t* iter81 = NULL;
+  size_t offset = 0;
+  size_t column = 1;
+  bool and = false;
+
+  // Base size of the query is 109 chars
+  // We then add space for the "?,"
+  char* statement = calloc(1, 109 + 2 * bundles_count + 2 * addresses_count +
```

Comment:
the logic of the dynamic statement creation should be in statements.c or another file